### PR TITLE
libbpf: 0.1.1 -> 0.4.0

### PR DESCRIPTION
**Pull request by akshatagarwl**
**DO NOT MERGE** Haven't run nixpkgs-review yet

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 131863` at c8cf9edc run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages_hardkernel_4_14.bcc</li>
    <li>linuxPackages_hardkernel_4_14.bpftrace</li>
    <li>linuxPackages_hardkernel_latest.bcc</li>
    <li>linuxPackages_hardkernel_latest.bpftrace</li>
  </ul>
</details>
<details>
  <summary>21 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/a60e509e1792c6ac0ae2958658eea3b4">linuxPackages.bcc</a> (linuxPackages_5_10.bcc)</li>
    <li><a href="https://gist.github.com/85ef3855bdfa12f760c0913103b788ff">linuxPackages.oci-seccomp-bpf-hook</a> (linuxPackages_5_10.oci-seccomp-bpf-hook)</li>
    <li><a href="https://gist.github.com/d6cfafa1d6de7a657334502d8324a443">linuxPackages_4_19.bcc</a></li>
    <li><a href="https://gist.github.com/e8e1e5a2b6eeec9308076fb2e83100b7">linuxPackages_5_12.bcc</a></li>
    <li><a href="https://gist.github.com/82a563fdb4e0e1c332e2067f5ca6ec43">linuxPackages_5_12.oci-seccomp-bpf-hook</a></li>
    <li><a href="https://gist.github.com/67ff3afab4f460e1369cd39908978c10">linuxPackages_5_13.bcc</a> (linuxPackages_latest.bcc)</li>
    <li><a href="https://gist.github.com/2014e1d9c61acc72b4cd478ef22f3c48">linuxPackages_5_13.oci-seccomp-bpf-hook</a> (linuxPackages_latest.oci-seccomp-bpf-hook)</li>
    <li><a href="https://gist.github.com/b760707912747621f833754419fd812b">linuxPackages_5_4.bcc</a></li>
    <li><a href="https://gist.github.com/30f4bbf5afbff2b6f6ab8810714d2a1e">linuxPackages_5_4.oci-seccomp-bpf-hook</a></li>
    <li><a href="https://gist.github.com/0b5e29fbc4bb3db7513a1f9eef9d0ad4">linuxPackages_hardened.bcc</a></li>
    <li><a href="https://gist.github.com/2ede43050efdaa36a0e7b9f25be804a3">linuxPackages_hardened.oci-seccomp-bpf-hook</a></li>
    <li><a href="https://gist.github.com/60c5c882051505a9dd304eee8a2acf38">linuxPackages_latest_hardened.bcc</a></li>
    <li><a href="https://gist.github.com/aa07b86e1f8a4be7bb3e524aef61033a">linuxPackages_latest_hardened.oci-seccomp-bpf-hook</a></li>
    <li><a href="https://gist.github.com/b33d0840d874ea933b621528c10d9308">linuxPackages_lqx.bcc</a></li>
    <li><a href="https://gist.github.com/6359d996dcd864f99ae6e196fb4e169b">linuxPackages_lqx.oci-seccomp-bpf-hook</a></li>
    <li><a href="https://gist.github.com/c65b3b7829411c75fd2c45907b80505d">linuxPackages_testing_bcachefs.bcc</a></li>
    <li><a href="https://gist.github.com/9d465f1c6095542551251e52c8f96299">linuxPackages_testing_bcachefs.oci-seccomp-bpf-hook</a></li>
    <li><a href="https://gist.github.com/278d91826102eddbc5f9f54b0d50deaa">linuxPackages_xanmod.bcc</a></li>
    <li><a href="https://gist.github.com/4c70e908fa351497e0b0f819e609d00f">linuxPackages_xanmod.oci-seccomp-bpf-hook</a></li>
    <li><a href="https://gist.github.com/0fcc3413eae19debced63b55e3713d73">linuxPackages_zen.bcc</a></li>
    <li><a href="https://gist.github.com/46e0eb8bedb978d1ab3fbba39c591aeb">linuxPackages_zen.oci-seccomp-bpf-hook</a></li>
  </ul>
</details>
<details>
  <summary>27 packages skipped due to time constraints:</summary>
  <ul>
    <li>dpdk</li>
    <li>linuxPackages-libre.bcc</li>
    <li>linuxPackages-libre.bpftrace</li>
    <li>linuxPackages-libre.oci-seccomp-bpf-hook</li>
    <li>pktgen (linuxPackages.pktgen)</li>
    <li>linuxPackages.bpftrace (linuxPackages_5_10.bpftrace)</li>
    <li>linuxPackages_4_14.bcc</li>
    <li>linuxPackages_4_14.bpftrace</li>
    <li>linuxPackages_4_19.bpftrace</li>
    <li>linuxPackages_4_4.bcc</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>knot-dns</li>
    <li>knot-resolver</li>
    <li>libbpf</li>
    <li>odp-dpdk (linuxPackages.odp-dpdk)</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`lgpl21` is a deprecated license, please check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/os-specific/linux/libbpf/default.nix:38:5:
```
   |
38 |     license = with licenses; [ lgpl21 /* or */ bsd2 ];
   |     ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/os-specific/linux/libbpf/default.nix [hunk 1]
```diff
@@ -1,36 +1,24 @@
...
-, fetchpatch
+{ lib
+, stdenv
+, fetchFromGitHub
+, pkg-config
+, libelf
+, zlib
 }:
 
 with builtins;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/libbpf/default.nix:9]: I don't think this is even used anywhere.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 131863` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
10 packages marked as broken and skipped:
</summary>
<ul>
<li>linuxPackages-libre.bcc</li>
<li>linuxPackages-libre.bpftrace</li>
<li>linuxPackages-libre.oci-seccomp-bpf-hook</li>
<li>linuxPackages_hardkernel_4_14.bcc</li>
<li>linuxPackages_hardkernel_4_14.bpftrace</li>
<li>linuxPackages_hardkernel_latest.bcc</li>
<li>linuxPackages_hardkernel_latest.bpftrace</li>
<li>linuxPackages_latest-libre.bcc</li>
<li>linuxPackages_latest-libre.bpftrace</li>
<li>linuxPackages_latest-libre.oci-seccomp-bpf-hook</li>
</ul>
</details>
<details>

  <summary>44 packages failed to build and are new build failures:</summary>
  <ul>
<li>linuxPackages.bcc: log was empty</li>
<li>(linuxPackages_5_10.bcc): log was empty</li>
<li>linuxPackages.bpftrace: log was empty</li>
<li>(linuxPackages_5_10.bpftrace): log was empty</li>
<li>linuxPackages.oci-seccomp-bpf-hook: log was empty</li>
<li>(linuxPackages_5_10.oci-seccomp-bpf-hook): log was empty</li>
<li>linuxPackages_4_14.bcc: log was empty</li>
<li>linuxPackages_4_14.bpftrace: log was empty</li>
<li>linuxPackages_4_19.bcc: log was empty</li>
<li>linuxPackages_4_19.bpftrace: log was empty</li>
<li>linuxPackages_4_4.bcc: log was empty</li>
<li>linuxPackages_4_4.bpftrace: log was empty</li>
<li>linuxPackages_4_9.bcc: log was empty</li>
<li>linuxPackages_4_9.bpftrace: log was empty</li>
<li>linuxPackages_5_12.bcc: log was empty</li>
<li>linuxPackages_5_12.bpftrace: log was empty</li>
<li>linuxPackages_5_12.oci-seccomp-bpf-hook: log was empty</li>
<li>linuxPackages_5_13.bcc: log was empty</li>
<li>(linuxPackages_latest.bcc): log was empty</li>
<li>linuxPackages_5_13.bpftrace: log was empty</li>
<li>(linuxPackages_latest.bpftrace): log was empty</li>
<li>linuxPackages_5_13.oci-seccomp-bpf-hook: log was empty</li>
<li>(linuxPackages_latest.oci-seccomp-bpf-hook): log was empty</li>
<li>linuxPackages_5_4.bcc: log was empty</li>
<li>linuxPackages_5_4.bpftrace: log was empty</li>
<li>linuxPackages_5_4.oci-seccomp-bpf-hook: log was empty</li>
<li>linuxPackages_hardened.bcc: log was empty</li>
<li>linuxPackages_hardened.bpftrace: log was empty</li>
<li>linuxPackages_hardened.oci-seccomp-bpf-hook: log was empty</li>
<li>linuxPackages_latest_hardened.bcc: log was empty</li>
<li>linuxPackages_latest_hardened.bpftrace: log was empty</li>
<li>linuxPackages_latest_hardened.oci-seccomp-bpf-hook: log was empty</li>
<li>linuxPackages_lqx.bcc: log was empty</li>
<li>linuxPackages_lqx.bpftrace: log was empty</li>
<li>linuxPackages_lqx.oci-seccomp-bpf-hook: log was empty</li>
<li>linuxPackages_testing_bcachefs.bcc: log was empty</li>
<li>linuxPackages_testing_bcachefs.bpftrace: log was empty</li>
<li>linuxPackages_testing_bcachefs.oci-seccomp-bpf-hook: log was empty</li>
<li>linuxPackages_xanmod.bcc: log was empty</li>
<li>linuxPackages_xanmod.bpftrace: log was empty</li>
<li>linuxPackages_xanmod.oci-seccomp-bpf-hook: log was empty</li>
<li>linuxPackages_zen.bcc: log was empty</li>
<li>linuxPackages_zen.bpftrace: log was empty</li>
<li>linuxPackages_zen.oci-seccomp-bpf-hook: log was empty</li>
  </ul>
</details>
<details>
<summary>
8 packages built:
</summary>
<ul>
<li>dpdk</li>
<li>knot-dns</li>
<li>knot-resolver</li>
<li>libbpf</li>
<li>odp-dpdk (linuxPackages.odp-dpdk)</li>
<li>pktgen (linuxPackages.pktgen)</li>
<li>spdk</li>
<li>suricata</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
dpdk:
<details>

/nix/store/4f0j2chl5399f3f2j6cdgid6cpxqh1lf-aws-c-event-stream-0.2.7
  /nix/store/4il2445aiq8f2v2kw0fddi9f2f1akfxb-aws-c-common-0.6.8
  /nix/store/6kd6xnkajkmh7d89lp332r0yjzyjmis4-aws-c-cal-0.5.11
  /nix/store/bzvv0wczbzd3hq0vs23qpp9vamgir05g-s2n-tls-1.0.1
  /nix/store/g0ifc64ndblfxvk17nzqp0nih6fd0qqs-aws-sdk-cpp-1.8.121
  /nix/store/ii3sy98bv82dyxvq3zl7ni9plvn1qq0d-nix-2.3.14-dev
  /nix/store/ka406i4zqa7l35sh5b9yf2hnx15bmfhl-boehm-gc-8.0.4-dev
  /nix/store/n2kwyncq0nf24004wqzjspsh9aa19q3c-busybox-static-x86_64-unknown-linux-musl-1.33.1
  /nix/store/n47fy4xk5vxqfhgiifq1alkpwsc41pmb-libsodium-1.0.18
  /nix/store/qfv2k2rwlsaqdab6fhr9h80c0y4jhq1n-aws-checksums-0.1.11
  /nix/store/rwps7cx0xmkfp3kymym3mgrby8m9nrnp-aws-c-io-0.9.1
  /nix/store/v3qadwn8mqik50x65qfbf1h3f67n32kr-nix-2.3.14-man
  /nix/store/v4yrsagbd6sf5cxdfi5514d6n1521b3m-nix-2.3.14
  /nix/store/vdi4bqhxd1jvy8yzvncwikg7kjrl3ff7-editline-1.17.1
  /nix/store/vvb07m15i3pgyjzglmyb8rf4z4zq5yvd-boehm-gc-8.0.4
copying path '/nix/store/n2kwyncq0nf24004wqzjspsh9aa19q3c-busybox-static-x86_64-unknown-linux-musl-1.33.1' from 'https://cache.nixos.org'...
copying path '/nix/store/v3qadwn8mqik50x65qfbf1h3f67n32kr-nix-2.3.14-man' from 'https://cache.nixos.org'...
copying path '/nix/store/4il2445aiq8f2v2kw0fddi9f2f1akfxb-aws-c-common-0.6.8' from 'https://cache.nixos.org'...
copying path '/nix/store/vvb07m15i3pgyjzglmyb8rf4z4zq5yvd-boehm-gc-8.0.4' from 'https://cache.nixos.org'...
copying path '/nix/store/vdi4bqhxd1jvy8yzvncwikg7kjrl3ff7-editline-1.17.1' from 'https://cache.nixos.org'...
copying path '/nix/store/6kd6xnkajkmh7d89lp332r0yjzyjmis4-aws-c-cal-0.5.11' from 'https://cache.nixos.org'...
copying path '/nix/store/qfv2k2rwlsaqdab6fhr9h80c0y4jhq1n-aws-checksums-0.1.11' from 'https://cache.nixos.org'...
copying path '/nix/store/ka406i4zqa7l35sh5b9yf2hnx15bmfhl-boehm-gc-8.0.4-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/n47fy4xk5vxqfhgiifq1alkpwsc41pmb-libsodium-1.0.18' from 'https://cache.nixos.org'...
copying path '/nix/store/bzvv0wczbzd3hq0vs23qpp9vamgir05g-s2n-tls-1.0.1' from 'https://cache.nixos.org'...
copying path '/nix/store/rwps7cx0xmkfp3kymym3mgrby8m9nrnp-aws-c-io-0.9.1' from 'https://cache.nixos.org'...
copying path '/nix/store/4f0j2chl5399f3f2j6cdgid6cpxqh1lf-aws-c-event-stream-0.2.7' from 'https://cache.nixos.org'...
copying path '/nix/store/g0ifc64ndblfxvk17nzqp0nih6fd0qqs-aws-sdk-cpp-1.8.121' from 'https://cache.nixos.org'...
copying path '/nix/store/v4yrsagbd6sf5cxdfi5514d6n1521b3m-nix-2.3.14' from 'https://cache.nixos.org'...
copying path '/nix/store/ii3sy98bv82dyxvq3zl7ni9plvn1qq0d-nix-2.3.14-dev' from 'https://cache.nixos.org'...
warning: unknown setting 'experimental-features'
warning: unknown setting 'extra-trusted-public-keys'
When evaluating attribute ‘dpdk’:
warning: unclear-gpl
`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/os-specific/linux/dpdk/default.nix:79:5:

```
   |
79 |     license = with licenses; [ lgpl21 gpl2 bsd2 ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
warning: unclear-gpl
`lgpl21` is a deprecated license, please check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/os-specific/linux/dpdk/default.nix:79:5:

```
   |
79 |     license = with licenses; [ lgpl21 gpl2 bsd2 ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
</details>

**Comment by vcunat** [comment]
Same change included in PR #131618

**Comment by humancalico** [comment]
@vcunat Thanks for referencing that.
Closing in favour of https://github.com/NixOS/nixpkgs/pull/131618

**Pull request closed without merging**